packages/splits-sdk-react/src/hooks/warehouse.ts

```diff
@@ -0,0 +1,107 @@
+import { Log } from 'viem'
+import { useCallback, useContext, useState } from 'react'
+import {
+  WarehouseWithdrawConfig,
+  WarehouseBatchWithdrawConfig,
+} from '@0xsplits/splits-sdk'
+
+import { SplitsContext } from '../context'
+import { ContractExecutionStatus, RequestError } from '../types'
+import { getSplitsClient } from '../utils'
+
+export const useWithdrawWarehouse = (): {
+  withdrawWarehouse: (
+    arg0: WarehouseWithdrawConfig,
+  ) => Promise<Log[] | undefined>
+  status?: ContractExecutionStatus
+  txHash?: string
+  error?: RequestError
+} => {
+  const context = useContext(SplitsContext)
+  const splitsClient = getSplitsClient(context).warehouse
+
+  if (!splitsClient) throw new Error('Invalid chain id for splits warehouse')
+
+  const [status, setStatus] = useState<ContractExecutionStatus>()
+  const [txHash, setTxHash] = useState<string>()
+  const [error, setError] = useState<RequestError>()
+
+  const withdrawWarehouse = useCallback(
+    async (argsDict: WarehouseWithdrawConfig) => {
+      try {
+        setStatus('pendingApproval')
```

Comment:
nit: we should move all these to enums. anyway, don't worry about it for this pr

Reply:
oh i guess at least we have a type defined so that protects it pretty well

Reply:
https://linear.app/splits/issue/PE-1951/move-react-sdk-strings-to-enumsconstants